### PR TITLE
Upgrade Stratos to 3.0.0

### PR DESCRIPTION
**Pull request by tammersaleh**
## Changes proposed in this pull request:

- Split out compilation and introduce caching (pipeline time reduced from ~1hr to ~10m)
- Track 3.x releases
- Nofify on 4.x releases (hopefully)
- Add the session secret
- Simplify the pipeline
- Remove manifest deprecations
- Remove map-routes in favor of `vars:`

## security considerations

None

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -162,20 +188,35 @@ resource_types:
...
 
+- name: s3-iam
+  type: docker-image
+  source:
+    repository: 18fgsa/s3-resource
+
 resources:
-- name: stratos-src
-  type: git
+- name: upstream
```
Comment (mxplusb) [inline ci/pipeline.yml]: Can you change this to `stratos-release` - it's less ambiguous. Also, adding `icon: github-circle` under `type` makes it easily identifiable as a Github construct.

File: ci/pipeline.yml [hunk 2]
```diff
@@ -162,20 +188,35 @@ resource_types:
...
+  type: github-release
+  source:
+    owner: cloudfoundry
+    repository: stratos
+    access_token: ((github-access-token))
+    # If you change this, change the one below as well.
+    tag_filter: '^(3\..*)'
+
+- name: next-upstream
```
Comment (mxplusb) [inline ci/pipeline.yml]: Can you change this to `next-stratos-release` - it's less ambiguous. Also, adding `icon: github-circle` under `type` makes it easily identifiable as a Github construct.

File: ci/pipeline.yml [hunk 3]
```diff
@@ -162,20 +188,35 @@ resource_types:
...
+    # If you change this, change the one above as well.
+    tag_filter: '^(4\..*)'
 
-- name: stratos-config
+- name: config
   type: git
   source:
     uri: https://github.com/18F/cg-deploy-stratos.git
-    branch: master
+    branch: tsaleh/track-upstream-releases
```
Comment (mxplusb) [inline ci/pipeline.yml]: Should this be `master`?

File: ci/precompile.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: node
+    tag: 10
```
Comment (mxplusb) [inline ci/precompile.yml]: This might be out of scope, but I think `lts` would be better so it's stable. Not sure on the compatibility, though.

Comment (tammersaleh) [reply ci/precompile.yml]: I'd rather not tackle that right now.  I'm not a node person, and this is what was there before the PR.

File: ci/pipeline.yml (hunk 3 above)
Comment (tammersaleh) [reply ci/pipeline.yml]: Actually, I need to pull it out into the secrets file so I can update after the PR is merged.

File: ci/precompile.yml (hunk 4 above)
Comment (mxplusb) [reply ci/precompile.yml]: Works for me.

**Pull request merged**